src/vs/workbench/contrib/debug/browser/debugService.ts

```diff
@@ -696,6 +698,7 @@ export class DebugService implements IDebugService {
 			}
 			this.endInitializingState();
 			this.cancelTokens(session.getId());
+			this.model.removeSession(session);
```

Comment:
Thanks for doing all this investigation with DebugSession! This isn't quite right though, because we want to keep the last session of each debug config so you can still browse past run output in the debug console. With this, if I run config A, then stop it and run config B, config A's output is gone and the debug console dropdown doesn't show up. And maybe there is some other problem from showing output in the repl from a session that is technically disposed. I can't remember exactly how it all works, I will look into it